src/audio/module_adapter/module_adapter.c

```diff
@@ -524,38 +531,59 @@ int module_adapter_copy(struct comp_dev *dev)
 	struct comp_buffer *sink;
 	struct processing_module *mod = comp_get_drvdata(dev);
 	struct module_data *md = &mod->priv;
+	struct comp_copy_limits c;
 	struct list_item *blist;
 	size_t size = MAX(mod->deep_buff_bytes, mod->period_bytes);
 	uint32_t min_free_frames = UINT_MAX;
 	int ret, i = 0;
 
-	/* get the least number of free frames from all sinks */
-	list_for_item(blist, &mod->sink_buffer_list) {
-		sink = container_of(blist, struct comp_buffer, sink_list);
-		min_free_frames = MIN(min_free_frames,
-				      audio_stream_get_free_frames(&sink->stream));
-	}
+	source = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
+	sink = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
```

Comment:
do I see it correctly, that these `source` and `sink` are only used in the `simple_copy` case below? If so, please move these assignments there

Reply:
first of all I think it's only `sink` that the compiler won't be able to figure out. So, yes, we'd have to initialise it to `NULL` - I think that's better than assigning irrelevant for the non-simple case values here to be overwritten there.